feat: replace node-fetch by fetch default API

PaulGOUX27 opened the pull request.

PaulGOUX27 committed:
feat!: replace node-fetch by fetch included with node

BREAKING CHANGE: requires node >= 18

PaulGOUX27 commented on package.json:
```diff
@@ -85,7 +85,6 @@
     "husky": "^4.3.0",
     "import-sort-cli": "^6.0.0",
     "import-sort-parser-typescript": "^6.0.0",
-    "import-sort-style-leboncoin": "^1.0.3",
```
Got 404 on yarn install. Just removed it as the repo is not under development

PaulGOUX27 commented:
Put PR into draft as the `node-fetch` is an ESM only package starting `3.0.0-beta.10`. I started to switch to `fetch` API included in node, but there is a bit of code to be added / tested. I will do it when I'm back

PaulGOUX27 commented:
@simonsmith I finished the PR. I removed a few things to make it run smoothly and fixed the release action by updating `semantic-release-action` to version 4. Merging it should create the new version.

simonsmith commented:
Amazing, let's try it

simonsmith commented:
Oh one question, usually `chore` does not create a new release. Would we want to have a `feat` or `fix` here?

https://github.com/pvdlg/conventional-changelog-metahub#commit-types

> Chores: Other changes that don't modify src or test files

PaulGOUX27 commented:
I changed to `feat`, but I changed the version in the `package.json` to the next major (as the minimal node version was updated). But on your link there is no major ... Hope it will work 🤞

simonsmith commented:
You should be able to revert the version change and semantic release will do it for us. So  the commit would have something like this in the description:

```
feat: do the change

BREAKING CHANGE: update node to whatever
```

That will trigger a `2.0.0` release

The breaking change part is detailed here for reference - https://www.conventionalcommits.org/en/v1.0.0/

PaulGOUX27 commented:
Alright, should be fine now. Thanks for sharing the convention, I didn't knew it exactly. Should we merge it now ?
